pkg/controller.v1beta1/trial/trial_controller_util.go

```diff
@@ -70,6 +70,15 @@ func (r *ReconcileTrial) UpdateTrialStatusCondition(instance *trialsv1beta1.Tria
 			msg := "Metrics are not available"
 			reason := TrialMetricsUnavailableReason
 
+			// If the type of metrics collector is Push, We should insert an unavailable value to Katib DB.
+			// We would retry reconciliation if some error occurs while we report unavailable metrics.
+			if instance.Spec.MetricsCollector.Collector.Kind == commonv1beta1.PushCollector {
+				if err := r.reportUnavailableMetrics(instance); err != nil {
+					logger.Error(err, "Failed to insert unavailable value to Katib DB")
+					return errReportMetricsFailed
```

Comment:
```suggestion
					return fmt.Errorf("%w: %w", errReportMetricsFailed, err)
```

We should wrap the error.

Reply:
LGTM